### PR TITLE
github workflow to run tests with PRs

### PR DESCRIPTION
**Pull request by iWebi**

### CODE REVIEW

File: .github/workflows/on-pr.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+on:
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java-version: [ 8 ]
```
Comment (zzantozz) [inline .github/workflows/on-pr.yml:11]: With a matrix build, the job name becomes "test (8)". It won't match the required commit status of "test". You can update the required commit status in the github settings repo. We had the same issue with the [blueflood repo definition](https://github.com/rax-maas/rax-maas-github-settings/blob/main/repos.yml#L26).

File: .github/workflows/on-pr.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: ${{ matrix.java-version }}
```
Comment (zzantozz) [inline .github/workflows/on-pr.yml:17]: I think you want a `cache: gradle` on here to avoid downloading the internet on every build.

**Review by zzantozz — APPROVED** [review]
Looks fine, but a couple of changes will be needed sooner or later.

File: .github/workflows/on-pr.yml (hunk 1 above)
Comment (iWebi) [reply .github/workflows/on-pr.yml:11]: 👍  will handle it in repo import config.

**Review by zzantozz — APPROVED** [review]

**Pull request merged**